File: artifacts/littlefs/decompose/src/lfs_file_t.c
typedef struct lfs_file
{
  struct lfs_file *next;
  uint16_t id;
  uint8_t type;
  lfs_mdir_t m;
  struct lfs_ctz
  {
    lfs_block_t head;
    lfs_size_t size;
  } ctz;
  uint32_t flags;
  lfs_off_t pos;
  lfs_block_t block;
  lfs_off_t off;
  lfs_cache_t cache;
  const struct lfs_file_config *cfg;
} lfs_file_t
/*** DEPENDENCIES:
typedef uint32_t lfs_block_t
----------------------------
typedef struct lfs_cache
{
  lfs_block_t block;
  lfs_off_t off;
  lfs_size_t size;
  uint8_t *buffer;
} lfs_cache_t
----------------------------
struct lfs_file_config
{
  void *buffer;
  struct lfs_attr *attrs;
  lfs_size_t attr_count;
}
----------------------------
typedef struct lfs_mdir
{
  lfs_block_t pair[2];
  uint32_t rev;
  lfs_off_t off;
  uint32_t etag;
  uint16_t count;
  bool erased;
  bool split;
  lfs_block_t tail[2];
} lfs_mdir_t
----------------------------
typedef uint32_t lfs_off_t
----------------------------
typedef uint32_t lfs_size_t
----------------------------
***/

File: artifacts/littlefs/translate/src/lfs_file_t.rs
use crate::*;
use std::clone::Clone;
use std::fs::File;


pub type lfs_block_t = u32;
pub type lfs_off_t = u32;
pub type lfs_size_t = u32;

#[derive(Clone)]
pub struct LfsCache {
    pub block: u32,
    pub off: u32,
    pub size: u32,
    pub buffer: Option<Vec<u8>>,
}

#[derive(Clone)]
pub struct LfsFileConfig {
    pub buffer: Option<Vec<u8>>,
    pub attrs: Option<Vec<LfsAttr>>,
    pub attr_count: crate::types::lfs_size_t,
}

#[derive(Clone)]
pub struct LfsMdir {
    pub pair: [lfs_block_t; 2],
    pub rev: u32,
    pub off: lfs_off_t,
    pub etag: u32,
    pub count: u16,
    pub erased: bool,
    pub split: bool,
    pub tail: [lfs_block_t; 2],
}

#[derive(Clone)]
pub struct LfsFile {
    pub next: Option<Box<LfsFile>>,
    pub id: u16,
    pub type_: u8,
    pub m: LfsMdir,
    pub ctz: LfsCtz,
    pub flags: u32,
    pub pos: lfs_off_t,
    pub block: lfs_block_t,
    pub off: lfs_off_t,
    pub cache: LfsCache,
    pub cfg: Option<Box<LfsFileConfig>>,
}

#[derive(Clone)]
pub struct LfsCtz {
    pub head: lfs_block_t,
    pub size: lfs_size_t,
}

